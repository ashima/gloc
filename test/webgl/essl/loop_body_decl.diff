--- loop_body_decl.true	2011-11-15 17:31:33.000000000 -0800
+++ loop_body_decl.out	2011-11-17 16:38:57.000000000 -0800
@@ -1 +1,10 @@
-error
+void main() {
+  int k = 0;
+  for (int i = 0; i < 10; i++) int i = k+i;/*  error, redecl no new scope*/
+  for (int i = 0; i < 10; i++) { int i = k+i; }/*  ok, compound nests*/
+  while (k > 0) int k = k--;/*  error, redecl no new scope*/
+  while (k > 0) { int k = k--; }/*  ok, compound nests*/
+  do int k = 1; while (k != 0);/*  ok, do always introduces scope*/
+  do { int k = 1; } while (k != 0);/*  ok, compound nests*/
+  gl_FragColor = vec4(float(k));
+}
