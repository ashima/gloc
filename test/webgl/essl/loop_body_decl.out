void main() {
  int k = 0;
  for (int i = 0; i < 10; i++) int i = k+i;/*  error, redecl no new scope*/
  for (int i = 0; i < 10; i++) { int i = k+i; }/*  ok, compound nests*/
  while (k > 0) int k = k--;/*  error, redecl no new scope*/
  while (k > 0) { int k = k--; }/*  ok, compound nests*/
  do int k = 1; while (k != 0);/*  ok, do always introduces scope*/
  do { int k = 1; } while (k != 0);/*  ok, compound nests*/
  gl_FragColor = vec4(float(k));
}
